bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/RenaultHandler.java

```diff
@@ -0,0 +1,113 @@
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RenaultHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Doug Culnane - Initial contribution
+ */
+@NonNullByDefault
+public class RenaultHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(RenaultHandler.class);
+
+    private @Nullable RenaultConfiguration config;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    public RenaultHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // This binding only polls status data automatically.
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        this.config = getConfigAs(RenaultConfiguration.class);
+
+        // Background initialization:
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(this::getStatus, 0, config.refreshInterval, TimeUnit.MINUTES);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+        super.dispose();
+    }
+
+    private void getStatus() {
+        MyRenaultHttpSession httpSession;
+        try {
+            httpSession = new MyRenaultHttpSession(this.config);
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            httpSession.updateCarData(this.config);
+            updateState(httpSession.getCar());
+        } catch (Exception e) {
+            httpSession = null;
+            logger.error("Error My Renault Http Session.", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void updateState(Car car) {
+        if (car.batteryLevel != null) {
+            updateState(CHANNEL_BATTERY_LEVEL, new DecimalType(car.batteryLevel));
+        }
+        if (car.hvacstatus != null) {
+            updateState(CHANNEL_HVAC_STATUS, (car.hvacstatus ? OnOffType.ON : OnOffType.OFF));
+        }
+        if (car.imageURL != null) {
+            updateState(CHANNEL_IMAGE, new StringType(car.imageURL));
+        }
+        if (car.gpsLatitude != null && car.gpsLongitude != null) {
+            updateState(CHANNEL_LOCATION,
+                    new PointType(new DecimalType(car.gpsLatitude), new DecimalType(car.gpsLongitude)));
+        }
+        if (car.odometer != null) {
+            updateState(CHANNEL_ODOMETER, new DecimalType(car.odometer));
```

Comment:
To use UoM, I suppose the value is in kilometers, so you case do it like that:
`updateState(CHANNEL_ODOMETER, new QuantityType<Length>(car.odometer.doubleValue, KILO(METRE)));`

Reply:
The JSON structure from which this parameter is collected is called "totalMileage" but my de_AT car value is km.  So I think km is the standard but I can not be sure for all locations.

Reply:
I guess I will set KM and let the US and UK refactor to the better system of units. ;-)

Reply:
Done.